src/models/thread.ts

```diff
@@ -504,8 +504,42 @@ export class Thread extends ReadReceipt<EmittedEvents, EventHandlerMap> {
         throw new Error("Unsupported function on the thread model");
     }
 
+    public getEventReadUpTo(userId: string, ignoreSynthesized?: boolean): string | null {
+        const isCurrentUser = userId === this.client.getUserId();
+        if (isCurrentUser && this.lastReply()) {
+            // If a thread last activity is prior the first read receipt sent in a thread
+            // we want to consider that this thread has been read
+            const beforeFirstThreadedReceipt = (this.lastReply()?.getTs() ?? 0) < this.room.oldestThreadedReceiptTs;
+            if (beforeFirstThreadedReceipt) {
+                return this.timeline.at(-1)?.getId() ?? null;
+            }
+        }
+
+        const readUpToId = super.getEventReadUpTo(userId, ignoreSynthesized);
+
+        // Checking whether the unthreaded read receipt for that user is more recent
+        // than the read receipt inside that thread.
+        if (isCurrentUser && this.lastReply()) {
```

Comment:
I'm having trouble following this comment, but have been unable to reword it in a way I'm happy with.

Why do we do this just for the current user? Why do we care about `lastReply()` (which is unused in the if-block)?

Reply:
You're right, we probably should apply that logic for all user and not only the current one.
Good spot

Reply:
I'm still not sure why we use `lastReply()` though? Should it be `readUpToId` instead?

Reply:
Hmm maybe we don't need to check `this.lastReply()` in the end.

If `readUpToId` is the last event of the timeline we should return `readUpToId` as no further computation is needed. However for all other cases we want to run the logic below.

And yeah that's slightly different than what is in the diff above. Will update